File: SegParte_Aula05/ClientServer/Backend/SimulacaoRepository.ts
import * as fs from 'fs';
import * as path from 'path';
import { Simulacao } from './Simulacao';

// Repository: Responsável pela Persistência (I/O)
export class SimulacaoRepository {

    private readonly nomeArquivo = path.join(__dirname, 'simulacoes_salvas_node.txt');

    public salvar(simulacao: Simulacao): void {
        const dadosSimulacao = simulacao.toString() + "\n---\n";

        try {
            // Usa fs.appendFileSync para adicionar ao final do arquivo de forma síncrona
            fs.appendFileSync(this.nomeArquivo, dadosSimulacao);
            console.log(`\n[REPOSITÓRIO] SUCESSO: Simulação salva em: ${this.nomeArquivo}`);
        } catch (error) {
            console.error('\n[REPOSITÓRIO] ERRO na camada de DADOS: Falha ao salvar no arquivo.', error);
            // Em uma API real, você lançaria o erro para a camada superior
        }
    }
}

File: SegParte_Aula05/MVC/SimulacaoRepository.java
package MVC;

import java.io.FileWriter;
import java.io.IOException;

// Repository/DAO: Responsável pela Persistência
public class SimulacaoRepository {

    private final String nomeArquivo = "simulacoes_salvas.txt";

    public void salvar(Simulacao simulacao) {
        // Lógica de Persistência (Poderia ser um DB, mas aqui é um arquivo)
        try (FileWriter fileWriter = new FileWriter(nomeArquivo, true)) {
            fileWriter.write(simulacao.toString() + "\n---\n");
            System.out.println("SUCESSO: Simulação salva no arquivo.");
        } catch (IOException e) {
            // Tratamento de Erro de Dados
            System.err.println("ERRO na camada de DADOS: Falha ao salvar no arquivo.");
            e.printStackTrace();
        }
    }
}